src/rest/app.py

```diff
@@ -0,0 +1,177 @@
+    if not app.openapi_schema:
+        openapi_schema = app.openapi()
+        openapi_schema["components"]["securitySchemes"] = {
+            "APIKeyHeader": {
+                "type": "apiKey",
+                "in": "header",
+                "name": "Authorization",
+                "description": "API Key needed to access this endpoint."
+            }
+        }
+        openapi_schema["security"] = [{"APIKeyHeader": []}]
+        app.openapi_schema = openapi_schema
+    return app.openapi_schema
+
+
+async def require_api_key(api_key: str = Security(api_key_header)):
+    """Dependency to require an API key for a route."""
+    config = CONFIG
+    if api_key not in config["api_keys"]:
+        LOGGER.warning(f"Unauthorized API key attempt: {api_key}")
+        raise HTTPException(
+            status_code=HTTP_401_UNAUTHORIZED, detail="Unauthorized")
+    return api_key
+
+
+@time_it
+@app.get("/", response_class=JSONResponse, dependencies=[Depends(require_api_key)])
+async def show_config():
+    """Returns the config of this service, excluding API keys."""
+    config_info = config.copy()
+    config_info.pop("api_keys", None)
+    return JSONResponse(content=config_info, status_code=200)
+
+
+@app.get("/health", response_class=PlainTextResponse, dependencies=[Depends(require_api_key)])
+async def health_check():
+    """Return health status."""
+    return "OK"
+
+
+@time_it
+@app.get("/transcriptions", dependencies=[Depends(require_api_key)])
+async def get_transcriptions():
+    """Get all transcriptions and their statuses."""
+    transcriptions = []
+    for file_name in DATA_HANDLER.get_all_status_filenames():
+        try:
+            file_name = file_name[:-5]  # remove .json
+            data = DATA_HANDLER.get_status_file_by_id(file_name)
+            transcriptions.append(
+                {
+                    "transcription_id": data["transcription_id"],
+                    "status": data["status"],
+                }
+            )
+        except Exception as e:
+            LOGGER.error(f"Error while reading status file {file_name}: {e}")
+            DATA_HANDLER.delete_status_file(file_name)
+    return JSONResponse(content=transcriptions, status_code=200)
+
+
+@time_it
+@app.get("/transcriptions/{transcription_id}", dependencies=[Depends(require_api_key)])
+async def get_transcriptions_id(transcription_id: str):
+    """Get the status of a transcription by ID."""
+    file = DATA_HANDLER.get_status_file_by_id(transcription_id)
+    if file:
+        return JSONResponse(content=file, status_code=200)
+    raise HTTPException(status_code=404, detail="Transcription ID not found")
+
+
+@time_it
+@app.post("/transcriptions", dependencies=[Depends(require_api_key)])
+async def post_transcription(
+    file: UploadFile = File(...),
+    language: str | None = Form("en"),
+    settings: str | None = Form('{}'),
+    model: str | None = Form("large-v3"),
```

Comment:
Is this the default or openapi hints? If its the ladder I think replacing this with tiny would make sense

Reply:
it is defaults